src/pages/SearchPage.js

```diff
@@ -45,69 +45,68 @@ const defaultProps = {
     reports: {},
 };
 
-class SearchPage extends Component {
-    constructor(props) {
-        super(props);
+function SearchPage(props) {
+    const {recentReports: initialRecentReports, personalDetails: initialPersonalDetails, userToInvite: initialUserToInvite} = OptionsListUtils.getSearchOptions(props.reports, props.personalDetails, '', props.betas);
 
-        Timing.start(CONST.TIMING.SEARCH_RENDER);
-        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    const [searchValue, setSearchValue] = useState('')
+    const [recentReports, setRecentReports] = useState(initialRecentReports)
+    const [personalDetails, setPersonalDetails] = useState(initialPersonalDetails)
+    const [userToInvite, setUserToInvite] = useState(initialUserToInvite)
 
-        this.searchRendered = this.searchRendered.bind(this);
-        this.selectReport = this.selectReport.bind(this);
-        this.onChangeText = this.onChangeText.bind(this);
-        this.debouncedUpdateOptions = _.debounce(this.updateOptions.bind(this), 75);
+    const updateOptions = useCallback(() => {
+        const {recentReports: localRecentReports, personalDetails: localPersonalDetails, userToInvite: localUserToInvite} = OptionsListUtils.getSearchOptions(
+            props.reports,
+            props.personalDetails,
+            searchValue.trim(),
+            props.betas,
+        );
 
-        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getSearchOptions(props.reports, props.personalDetails, '', props.betas);
+        setUserToInvite(localUserToInvite)
+        setRecentReports(localRecentReports)
+        setPersonalDetails(localPersonalDetails)
+    }, [props.reports, props.personalDetails, searchValue, props.betas])
 
-        this.state = {
-            searchValue: '',
-            recentReports,
-            personalDetails,
-            userToInvite,
-        };
-    }
+    const debouncedUpdateOptions =_.debounce(updateOptions, 75);
 
-    componentDidUpdate(prevProps) {
-        if (_.isEqual(prevProps.reports, this.props.reports) && _.isEqual(prevProps.personalDetails, this.props.personalDetails)) {
-            return;
-        }
-        this.updateOptions();
-    }
+    useEffect(() => {
+        Timing.start(CONST.TIMING.SEARCH_RENDER);
+        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    },[])
 
-    onChangeText(searchValue = '') {
-        this.setState({searchValue}, this.debouncedUpdateOptions);
-    }
+    useEffect(() => {
+        updateOptions();
+    },[props.reports, props.personalDetails])
```

Comment:
I think this `useEffect` should have only this function as a dependency. It is logically. Finally, it uses the same deps.
```suggestion
    },[updateOptions])
```